src/abstractions/src/functions/KernelFunction.ts

```diff
@@ -113,30 +97,26 @@ export abstract class KernelFunction<
 }
 
 export const kernelFunction = <
-  Schema extends JsonSchema | undefined = undefined,
-  Result = unknown,
-  Args = Schema extends JsonSchema ? FromSchema<Schema> : undefined,
+  PARAMETERS extends AIFunctionParameterMetadata = AIFunctionParameterMetadata,
+  SCHEMA = FromSchema<PARAMETERS>,
 >(
-  fn: Fn<Result, Args>,
-  metadata: KernelFunctionMetadata<Schema>
-): KernelFunction<Schema, Result, Args> => {
-  return new (class extends KernelFunction<Schema, Result, Args> {
-    constructor() {
-      super({ metadata });
+  fn: (args?: SCHEMA, kernel?: Kernel) => unknown,
+  metadata: KernelFunctionMetadata<PARAMETERS>
+): KernelFunction<PARAMETERS, SCHEMA> => {
+  return new (class extends KernelFunction<PARAMETERS, SCHEMA> {
+    public constructor() {
+      super(metadata);
     }
 
-    protected override invokeStreamingCore<T>(): AsyncGenerator<T> {
+    protected override invokeStreamingCore(): AsyncGenerator<unknown> {
```

Comment:
Stub for future?

Reply:
KernelFunction has an invokeStreaming in addition to invoke. Some functions like `KernelFunctionFromPrompt` actually use this.